File: beichen-erp-server/src/main/java/com/beichen/erp/customer/entity/Customer.java
package com.beichen.erp.customer.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("customer")
public class Customer {

    @TableId(type = IdType.AUTO)
    private Long id;

    private String code;

    private String name;

    private String contact;

    private String phone;

    private String address;

    private Integer creditPeriod;

    private Integer creditPeriodMonths;

    private BigDecimal creditLimit;

    private BigDecimal receivableBalance;

    private BigDecimal prepaidBalance;

    private Integer status;

    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private Long companyId;

    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;
}

File: beichen-erp-web/src/api/customer.ts
import request from '@/utils/request'

export interface Customer {
  id?: number
  code: string
  name: string
  contact?: string
  phone?: string
  address?: string
  creditPeriod?: number
  creditPeriodMonths?: number
  creditLimit?: number
  receivableBalance?: number
  prepaidBalance?: number
  status: number
  remark?: string
}

export interface PageResult<T> {
  records: T[]
  total: number
  current: number
  size: number
}

export function getCustomerPage(params: any) {
  return request.get<unknown, PageResult<Customer>>('/inventory/customer/page', { params })
}

export function listCustomers() {
  return request.get<unknown, Customer[]>('/inventory/customer/list')
}

export function getCustomer(id: number) {
  return request.get<unknown, Customer>(`/inventory/customer/${id}`)
}

export function createCustomer(data: Customer) {
  return request.post<unknown, void>('/inventory/customer', data)
}

export function updateCustomer(data: Customer) {
  return request.put<unknown, void>('/inventory/customer', data)
}

export function updateCustomerStatus(id: number, status: number) {
  return request.put<unknown, void>(`/inventory/customer/${id}/status`, null, { params: { status } })
}
